torch/utils/cpp_extension.py

```diff
@@ -828,6 +828,39 @@ def CUDAExtension(name, sources, *args, **kwargs):
                 cmdclass={
                     'build_ext': BuildExtension
                 })
+
+    Compute capabilities:
+
+    By default the extension will be compiled to run on all archs of the cards visible during the
+    building process of the extension. If down the road a new card is installed the extension may
+    need to be recompiled.
+
+    This default may also clamp a higher arch to a lower one that has the same binary compatibility
+    if pytorch was compiled with that lower one. e.g. ``sm_86`` could be clamped down to `sm_80`.
+    For a variety of technical reasons the distributed pytorch binary doesn't build against the full
+    range of computer capabilities, e.g. it includes only `sm_60` and `sm_70`,but not `sm_61` and
+    `sm_75`. The not included ones are binary compatible with the included ones, but you might not
+    be getting the best performance.
```

Comment:
This is inaccurate, but explaining it fully might be more confusing to readers than it's worth.  I think we can delete this paragraph without losing actionable guidance (the actionable guidance is in other paragraphs).

Reply:
1. I'd love to have the full understanding, so yes please to go into more details on slack - thank you!
2. I think it's still worth to at least add a note that clamping may be performed - perhaps w/o trying to explain why. I found it very confusing when I started debugging this at why I was getting sm_80, when my card was reporting sm_86. But I trust your judgment at how to provide the best user-experience

Reply:
Perhaps leaving just:

```
This default may also clamp a higher arch to a lower one that has the same binary compatibility
    if pytorch was compiled with that lower one. e.g. ``sm_86`` could be clamped down to `sm_80`.
```
?

Reply:
In the absence of PTX, clamping _may not_ be performed.  The fact that 8.0 fully-compiled binary code (SASS) happens to work on 8.6 is not guaranteed by Cuda afaik, and in general we should not rely on a binary compiled for `X.0` being usable on any `X.Y` unless we compiled with `X.0+PTX`.

Reply:
OK, so this is no longer about an inaccurate description, but you're suggesting to change the functionality, as currently clamping is performed regardless.

Reply:
I recommended adding `+PTX` to the highest compiled arch when clamping https://github.com/pytorch/pytorch/pull/48891/files#r537800878.

Reply:
done